Add aten::slice NNAPI converter

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434
* #59365
* **#59364**
* #58543
* #60885
* #58541
* #58540
* #58539

Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

Commit:
add aten:;slice

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "add aten:;slice"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "add aten:;slice"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

Commit:
Update on "add aten:;slice"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

dreiss commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -971,6 +974,69 @@ def add_flatten(self, node):
+
+        if start_value < 0:
+            start_value += in_oper.shape[dim_value]
+        elif start_value == sys.maxsize:
```
Why would this happen?

axitkhurana replied:
* https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorIndexing.h#L18
* https://github.com/pytorch/pytorch/blob/master/test/cpp/api/tensor_indexing.cpp#L45

None ends up translating to INDEX_MAX

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

Commit:
Update on "Add aten::slice NNAPI converter"


Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).
